src/main/groovy/edu/oregonstate/mist/beaverbus/RideSystemsDAO.groovy

```diff
@@ -0,0 +1,149 @@
+package edu.oregonstate.mist.beaverbus
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties
+import com.fasterxml.jackson.annotation.JsonProperty
+import com.fasterxml.jackson.core.type.TypeReference
+import com.fasterxml.jackson.databind.JsonMappingException
+import com.fasterxml.jackson.databind.ObjectMapper
+import groovy.transform.CompileStatic
+import groovy.transform.ToString
+import org.apache.http.client.HttpClient
+import org.apache.http.client.methods.HttpGet
+import org.apache.http.util.EntityUtils
+
+import javax.ws.rs.core.UriBuilder
+
+@CompileStatic
+class RideSystemsDAO {
+    HttpClient httpClient
+    URL baseURL
+    String apiKey
+
+    ObjectMapper mapper = new ObjectMapper()
+
+    List<RouteWithSchedule> getRoutesForMapWithScheduleWithEncodedLine() {
+        def url = UriBuilder.fromUri(baseURL.toURI())
+                .path("GetRoutesForMapWithScheduleWithEncodedLine")
+                .queryParam("ApiKey", apiKey)
+                .build()
+        def resp = httpClient.execute(new HttpGet(url))
+        // XXX check http code?
+        // XXX catch IO Exception?
+        def body = EntityUtils.toString(resp.entity)
+        try {
+            (List<RouteWithSchedule>) mapper.readValue(body,
+                    new TypeReference<List<RouteWithSchedule>>() {})
+        } catch (JsonMappingException exc) {
+            // Catch and wrap the jsonmappingexception.
+            // Dropwizard catches and eats any jsonmappingexception thrown during a request and
+            // returns an unhelpful error message, probably assuming that the exception was caused
+            // by a syntax error in the request body, which is not the case here
+            throw new Exception(exc)
+        }
+    }
+
+    List<Vehicle> getMapVehiclePoints(Integer routeID) {
+        def builder = UriBuilder.fromUri(baseURL.toURI()).path("GetMapVehiclePoints")
+        builder.queryParam("ApiKey", apiKey)
+        if (routeID != null) {
```

Comment:
same here?

Reply:
No, `!routeID` would also ignore `routeID=0`, which doesn't seem right.

...Although it amounts to the same thing because we pass routeID directly to RideSystems, and _they_ ignore `routeID=0`. Hmm.